components/automate-cli/cmd/chef-automate/summary.go

```diff
@@ -0,0 +1,418 @@
+	}
+
+	health := fmt.Sprint(ServiceHealth["status"])
+
+	args = []string{
+		"-s",
+		fmt.Sprintf("%s/census", habUrl),
+		"--header",
+		fmt.Sprintf("'Authorization: Bearer %s'", authToken),
+		"-k",
+	}
+	censusData, err := executeCmd(sshUtil, "curl", args)
+	if err != nil {
+		return BeStatusValue{}, err
+	}
+
+	populationData := censusData["census_groups"].(map[string]interface{})[service+".default"].(map[string]interface{})["population"].(map[string]interface{})[fmt.Sprintf("%s", memeber_id)]
+	getrole, ok := populationData.(map[string]interface{})
+	role := "Unknown"
+	if ok {
+		isleader, _ := strconv.ParseBool(fmt.Sprint(getrole["leader"]))
+		isfollower, _ := strconv.ParseBool(fmt.Sprint(getrole["follower"]))
+		if isleader {
+			role = "Leader"
+		} else if isfollower {
+			role = "Follower"
+		}
+	}
+
+	t := time.Unix(int64(startingTime), 0)
+	duration := t.Sub(time.Unix(0, 0))
+
+	// Format the duration as "1d 148h 43m 31s"
+	formatted := fmt.Sprintf("%dd %dh %dm %ds", int(duration.Hours())/24, int(duration.Hours())%24, int(duration.Minutes())%60, int(duration.Seconds())%60)
+
+	return BeStatusValue{
+		serviceName: serviceType,
+		ipAddress:   ip,
+		health:      health,
+		process:     fmt.Sprintf("%s (pid: %s)", serviceState, servicePid),
+		upTime:      formatted,
+		role:        role,
+	}, nil
+}
+
+func (ss *Summary) getFEStatus(sshUtil SSHUtil, script string, ip string, serviceType string) FeStatusValue {
+	osStatus := ss.opensearchStatusInFE(sshUtil)
+	_, err := sshUtil.connectAndExecuteCommandOnRemote(script, true)
+	status := "OK"
+	if err != nil {
+		_, ok := err.(*exec.ExitError)
+		if ok {
+			status = "ERROR"
+		} else {
+			status = "WARN"
+		}
+	}
+	return FeStatusValue{
+		serviceName: serviceType,
+		ipAddress:   ip,
+		status:      status,
+		Opensearch:  osStatus,
+	}
+}
+
+func (ss *Summary) opensearchStatusInFE(sshUtil SSHUtil) string {
+	arg := []string{
+		"-s",
+		clusterOSHealth,
+	}
+	curlOSStatusScript := GenerateOriginalAutomateCLICommand(
+		&cobra.Command{
+			Use: "curl",
+		}, arg)
+	osStatusJson, _ := sshUtil.connectAndExecuteCommandOnRemote(curlOSStatusScript, true)
+
+	body := []byte(osStatusJson)
+	esHealthData := make(map[string]json.RawMessage)
+
+	err := json.Unmarshal(body, &esHealthData)
+	if err != nil {
+		return "Unknown"
+	}
+
+	return fmt.Sprintf("%s (Active: %s)", esHealthData["status"], esHealthData["active_shards_percent_as_number"])
+}
+
+func executeCmd(sshUtil SSHUtil, cmd string, args []string) (map[string]interface{}, error) {
+	script := GenerateOriginalAutomateCLICommand(&cobra.Command{
+		Use: "curl",
+	}, args)
+	output, err := sshUtil.connectAndExecuteCommandOnRemote(script, true)
+	if err != nil {
+		return nil, err
+	}
+
+	var body map[string]interface{}
+	err = json.Unmarshal([]byte(output), &body)
+	if err != nil {
+		return nil, err
+	}
+	return body, nil
+}
+
+func (ss *Summary) checkIPAddresses() ([]string, []string, []string, []string, *list.List) {
```

Comment:
This function is too large. Please break it into multiple smaller functions.

Reply:
scope of func is to check and get ip-addresses by services

Reply:
done